### PR TITLE
sentry-native/0.4.12: Bump openssl and libcurl version

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **sentry-native/0.4.12**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — DISMISSED** [review]

File: recipes/sentry-native/all/conanfile.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def requirements(self):
                 self.requires("breakpad/cci.20210521")
         if self.options.qt:
             self.requires("qt/5.15.2")
-            self.requires("openssl/1.1.1k")
+            self.requires("openssl/1.1.1l")
```
Comment (madebr) [inline recipes/sentry-native/all/conanfile.py:122]: I'm curious. Does 3.0.0 work?

Comment (MartinDelille) [reply recipes/sentry-native/all/conanfile.py:122]: It seems to work locally but I didn't test further.

Comment (madebr) [reply recipes/sentry-native/all/conanfile.py:122]: Thanks. I was wondering about the API compatibility.

**Review by uilianries — APPROVED** [review]

File: recipes/sentry-native/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/sentry-native/all/conanfile.py:126]: Compatibility is really freaking close, someone contribute to a project I am apart of https://github.com/Thalhammer/jwt-cpp 

One broken const https://github.com/Thalhammer/jwt-cpp/commit/17fdd26d97dec5b77c03ccef9057b343894e54dd

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**